src/inc/corcompile.h

```diff
@@ -905,22 +905,16 @@ class CorProfileData
 
     struct CORBBTPROF_TOKEN_INFO *  GetTokenFlagsData(SectionFormat section)
     {
-        if (this == NULL)
```

Comment:
I do not see what makes these safe to delete.

Reply:
I was relying on tests to validate these changes. This combination worked for various configurations (and passed 41 CI checks). I can all null checks on all callsites, if it is not considered safe.

Reply:
I think that would be better.

E.g. It think `Module::GetProfileData()` in `Module::ExpandAll` can return NULL that this check was protecting against. We may be missing coverage for these paths in the CI.